src/planet-a/IcoSphereTree.h

```diff
@@ -67,8 +81,70 @@ static constexpr int gc_icosahedronVertCount = 12;
 static constexpr std::uint8_t gc_triangleMaskSubdivided = 0b0001;
 //static constexpr std::uint8_t gc_triangleMaskChunked    = 0b0010;
 
-// An icosahedron with subdividable faces
-// it starts with 20 triangles, and each face can be subdivided into 4 more
+/**
+ * A simple 1D translation and scale
+ */
+struct TriangleSideTransform
+{
+    float m_translation{0.0f};
+    float m_scale{1.0f};
+};
+
+class IcoSphereTreeObserver
+{
+public:
+    using Handle_T = std::list<std::weak_ptr<IcoSphereTreeObserver>>::iterator;
```

Comment:
An iterator to a  weak_ptr?

Why not just std::shared_ptr, or std::weak_ptr, as a handle? Is knowing the specific position in the list important for a handle? How are you dealing with iterator invalidation?

Reply:
These handles are used to delete an observer from an IcoSphereTree's list, with 
```void IcoSphereTree::event_remove(IcoSphereTreeObserver::Handle_T)```
though this function implemented yet.